File: iosApp/iosApp/server_config/Saml2Config.swift
import Foundation

/**
 * From: https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/home/saml2-login/saml2.config.ts
 */
struct Saml2Config: Decodable {
    var identityProviderName: String? = nil
    var buttonLabel: String? = nil
    var passwordLoginDisabled: Bool = false
    var enablePassword: Bool = false

    enum CodingKeys: String, CodingKey {
        case identityProviderName
        case buttonLabel = "button-label"
        case passwordLoginDisabled
        case enablePassword
    }
}

File: core/model/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/android/model/server_config/Saml2Config.kt
package de.tum.informatics.www1.artemis.native_app.android.model.server_config

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * From: https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/home/saml2-login/saml2.config.ts
 */
@Serializable
data class Saml2Config(
    val identityProviderName: String? = null,
    @SerialName("button-label")
    val buttonLabel: String? = null,
    val passwordLoginDisabled: Boolean = false,
    val enablePassword: Boolean = false
)